force_torque_sensor_broadcaster/src/force_torque_sensor_broadcaster.cpp

```diff
@@ -148,15 +211,71 @@ controller_interface::return_type ForceTorqueSensorBroadcaster::update_and_write
   {
     params_ = param_listener_->get_params();
   }
-  if (realtime_publisher_ && realtime_publisher_->trylock())
+  wrench_raw_.header.stamp = time;
+  force_torque_sensor_->get_values_as_message(wrench_raw_.wrench);
+  this->apply_sensor_offset(params_, wrench_raw_);
+  this->apply_sensor_multiplier(params_, wrench_raw_);
+
+  if (realtime_raw_publisher_ && realtime_raw_publisher_->trylock())
   {
-    realtime_publisher_->msg_.header.stamp = time;
-    force_torque_sensor_->get_values_as_message(realtime_publisher_->msg_.wrench);
-    this->apply_sensor_offset(params_, realtime_publisher_->msg_);
-    this->apply_sensor_multiplier(params_, realtime_publisher_->msg_);
-    realtime_publisher_->unlockAndPublish();
+    realtime_raw_publisher_->msg_.header.stamp = time;
+    realtime_raw_publisher_->msg_.wrench = wrench_raw_.wrench;
+    realtime_raw_publisher_->unlockAndPublish();
   }
 
+  // Filter sensor data
+  auto filtered = filter_chain_->update(wrench_raw_, wrench_filtered_);
+  if (!filtered)
+  {
+    wrench_filtered_.wrench.force.x = std::numeric_limits<double>::quiet_NaN();
+    wrench_filtered_.wrench.force.y = std::numeric_limits<double>::quiet_NaN();
+    wrench_filtered_.wrench.force.z = std::numeric_limits<double>::quiet_NaN();
+    wrench_filtered_.wrench.torque.x = std::numeric_limits<double>::quiet_NaN();
+    wrench_filtered_.wrench.torque.y = std::numeric_limits<double>::quiet_NaN();
+    wrench_filtered_.wrench.torque.z = std::numeric_limits<double>::quiet_NaN();
+  }
+
+  if (realtime_filtered_publisher_ && realtime_filtered_publisher_->trylock())
+  {
+    realtime_filtered_publisher_->msg_.header.stamp = time;
+    realtime_filtered_publisher_->msg_.wrench = wrench_filtered_.wrench;
+    realtime_filtered_publisher_->unlockAndPublish();
+  }
+
+  for (const auto & publisher : wrench_additional_frames_publishers_)
+  {
+    try
+    {
+      if (filtered)
+      {
+        auto transform = tf_buffer_->lookupTransform(
+          publisher->msg_.header.frame_id, params_.frame_id, tf2::TimePointZero);
+        tf2::doTransform(wrench_filtered_, publisher->msg_, transform);
+      }
+      else
+      {
+        throw tf2::TransformException("cannot transform, filtered message is invalid");
+      }
```

Comment:
I'm not a big fan of having tf inside the RT loops. If this is the case, I would ask you to time it and share the results on how much time it consumes 1. static transforms 2. dynamic transforms